syspagegen/syspagegen.c

```diff
@@ -0,0 +1,822 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Tool to generate syspage based on plo scripts
+ *
+ * Copyright 2022 Phoenix Systems
+ *
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <sys/stat.h>
+#include <sys/queue.h>
+
+#include "syspage.h"
+
+
+#define ALIGN_ADDR(addr, align) (align ? ((addr + (align - 1)) & ~(align - 1)) : addr)
+#define CAST_SYSPTR(type, ptr)  ((type)(sysgen_common.buff + ptr - (sysgen_common.pkernel + sysgen_common.offs)))
+
+/* Reserve +1 for terminating NULL pointer in conformance to C standard */
+#define SIZE_CMD_ARGV     (10 + 1)
+#define SIZE_CMD_ARG_LINE 181
+
+
+struct phfs_alias_t {
+	char name[32];
+	addr_t addr;
+	size_t size;
+
+	LIST_ENTRY(phfs_alias_t) ptrs;
+};
+
+
+typedef struct {
+	const char name[12];
```

Comment:
const string will be assigned anyway, we don't need to reserve storage for these strings

```suggestion
	const char *name;
```

Reply:
Done